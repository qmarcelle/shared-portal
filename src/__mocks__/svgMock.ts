import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
