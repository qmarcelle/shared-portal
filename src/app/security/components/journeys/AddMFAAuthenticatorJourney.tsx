import { ErrorDisplaySlide } from '@/components/composite/ErrorDisplaySlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Divider } from '@/components/foundation/Divider';
import { QRCodeGenerator } from '@/components/foundation/QRCodeGenerator';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';
import { AppProg } from '@/models/app_prog';
import { ComponentDetails } from '@/models/component_details';
import { useEffect, useRef, useState } from 'react';
import { MfaDeviceType } from '../../models/mfa_device_type';
import { VerifyMfaResponse } from '../../models/verify_mfa_devices';
import { useSecuritySettingsStore } from '../../stores/security_settings_store';

export const AddMFAAuthenticatorJourney = ({
  changePage,
  pageIndex,
}: ModalChildProps) => {
  const {
    updateMfaDevice,
    verifyMfaDevice,
    verifyMfaResult,
    updatedMfaResult,
    resetVerifyMfaError,
  } = useSecuritySettingsStore();
  const [confirmCode, setConfirmCode] = useState('');
  const { dismissModal } = useAppModalStore();

  const initialized = useRef(false);
  useEffect(() => {
    (async () => {
      if (!initialized.current) {
        initialized.current = true;
        try {
          await updateMfaDevice(MfaDeviceType.authenticator);
        } catch (err) {
          // Change to Error page
          changePage!(2, false);
        }
      }
    })();
  });

  const submitCode = async () => {
    try {
      // Do API call for submit code
      const response: ComponentDetails<VerifyMfaResponse> | undefined =
        await verifyMfaDevice(MfaDeviceType.authenticator, confirmCode);
      if (response?.state == AppProg.success) {
        changePage?.(1, false);
      }
    } catch (err) {
      // Change to Error page
      changePage!(2, false);
    }
  };

  const updateSecurityCode = (value: string) => {
    setConfirmCode(value);
    if (verifyMfaResult?.errors.length) {
      resetVerifyMfaError();
    }
  };

  const pages = [
    <InputModalSlide
      key={0}
      label="Authenticator App Setup"
      subLabel="Scan the QR Code or manually type the secret key into your authenticator app then enter the security code."
      actionArea={
        <Column className="items-center">
          <div style={{ minWidth: 132, height: 132 }}>
            <QRCodeGenerator value={updatedMfaResult?.result?.keyUri ?? ''} />
          </div>
          <Spacer size={24} />
          <TextBox className="body-1 font-bold" text={'Secret Key:'} />
          <Spacer size={8} />
          <TextBox
            text={updatedMfaResult?.result?.secret ?? ''}
            type="body-1"
          />
          <Spacer size={24} />
          <Divider />
          <Spacer size={24} />
          <TextField
            valueCallback={(val) => updateSecurityCode(val)}
            label={'Enter Security Code'}
            errors={verifyMfaResult?.errors}
          ></TextField>
          <Spacer size={24} />
        </Column>
      }
      cancelCallback={() => dismissModal()}
      nextCallback={confirmCode.length > 5 ? () => submitCode() : undefined}
    />,

    <SuccessSlide
      key={1}
      label="Authenticator App Setup Complete"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="The next time you login, you'll generate a code through your authenticator app."
          />
          <Spacer size={32} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
    <ErrorDisplaySlide
      key={2}
      label="Try Again Later"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="Oops! We're sorry. Something went wrong. Please try again."
          />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
