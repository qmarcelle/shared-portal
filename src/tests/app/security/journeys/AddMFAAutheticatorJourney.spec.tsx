import { AddMFAAuthenticatorJourney } from '@/app/security/components/journeys/AddMFAAuthenticatorJourney';
import { AppModal, useAppModalStore } from '@/components/foundation/AppModal';
import { createAxiosErrorForTest } from '@/tests/test_utils';
import '@testing-library/jest-dom';
import {
  RenderResult,
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { mockedAxios } from '../../../__mocks__/axios';

const renderUI = () => {
  return render(<AppModal />);
};

jest.mock('../../../../utils/server_session', () => ({
  getServerSideUserId: jest.fn(() => Promise.resolve('test1234')),
}));

describe('Authenticator App Setup', () => {
  let component: RenderResult;
  const showAppModal = useAppModalStore.getState().showAppModal;
  const dismissAppModal = useAppModalStore.getState().dismissModal;
  beforeEach(() => {
    dismissAppModal();
    mockedAxios.post.mockResolvedValueOnce({
      data: {
        data: {
          message: 'Phone already registered.',
          deviceType: 'SMS',
          deviceStatus: 'ACTIVATION_REQUIRED',
          createdAt: '2024-02-09T12:40:33.554Z',
          updatedAt: '2024-02-09T12:40:33.554Z',
          phone: '11111111111',
          email: 'thomas@abc.com',
          secret: 'ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ',
          keyUri:
            'otpauth://totp/thomas@abc.com?secret=ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ',
        },
      },
    });
    component = renderUI();
    showAppModal({ content: <AddMFAAuthenticatorJourney /> });
  });

  it('should render the screens correctly', async () => {
    // Init Screen is rendered correctly
    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: 'Authenticator App Setup' }),
      ).toBeVisible();
      expect(screen.getByText('Secret Key:')).toBeVisible();
      expect(
        screen.getByText('ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ'),
      ).toBeVisible();
    });
    expect(component.baseElement).toMatchSnapshot();

    mockedAxios.post.mockRejectedValue(
      createAxiosErrorForTest({
        errorObject: {
          data: { errorCode: 'INVALID_OTP' },
        },
        status: 400,
      }),
    );

    const securityCode = screen.getByLabelText(/Enter Security Code/i);
    await userEvent.type(securityCode, '123456');
    fireEvent.click(screen.getByRole('button', { name: /confirm/i }));
    await waitFor(() => {
      expect(
        screen.getByText(
          'There is a problem with the security code. Try re-entering or resending the code.',
        ),
      ).toBeVisible();
    });

    mockedAxios.post.mockResolvedValueOnce({
      data: {
        data: {
          deviceStatus: 'ACTIVE',
          createdAt: '2024-02-09T12:40:33.554Z',
          updatedAt: '2024-02-09T12:40:33.554Z',
          phone: '11111111111',
          email: 'thomas@abc.com',
        },
      },
    });

    const codeEntryInput = screen.getByLabelText(/Enter Security Code/i);
    await userEvent.type(codeEntryInput, '123456');
    await waitFor(() => {
      expect(
        screen.queryByText(
          'There is a problem with the security code. Try re-entering or resending the code.',
        ),
      ).not.toBeInTheDocument();
    });
    fireEvent.click(screen.getByRole('button', { name: /confirm/i }));
    // Success screen rendered correctly
    await waitFor(() => {
      expect(
        screen.getByRole('heading', {
          name: 'Authenticator App Setup Complete',
        }),
      ).toBeVisible();
    });
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render the screens without error when we close the modal previously with error', async () => {
    // Init Screen is rendered correctly
    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: 'Authenticator App Setup' }),
      ).toBeVisible();
      expect(screen.getByText('Secret Key:')).toBeVisible();
      expect(
        screen.getByText('ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ'),
      ).toBeVisible();
    });
    expect(component.baseElement).toMatchSnapshot();

    mockedAxios.post.mockRejectedValue(
      createAxiosErrorForTest({
        errorObject: {
          data: { errorCode: 'INVALID_OTP' },
        },
        status: 400,
      }),
    );

    const securityCode = screen.getByLabelText(/Enter Security Code/i);
    await userEvent.type(securityCode, '123456');
    fireEvent.click(screen.getByRole('button', { name: /confirm/i }));
    await waitFor(() => {
      expect(
        screen.getByText(
          'There is a problem with the security code. Try re-entering or resending the code.',
        ),
      ).toBeVisible();
    });
    fireEvent.click(screen.getByAltText('close'));
    await waitFor(() => {
      expect(
        screen.queryByText('Authenticator App Setup'),
      ).not.toBeInTheDocument();
    });
    mockedAxios.post.mockResolvedValueOnce({
      data: {
        data: {
          message: 'Phone already registered.',
          deviceType: 'SMS',
          deviceStatus: 'ACTIVATION_REQUIRED',
          createdAt: '2024-02-09T12:40:33.554Z',
          updatedAt: '2024-02-09T12:40:33.554Z',
          phone: '11111111111',
          email: 'thomas@abc.com',
          secret: 'ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ',
          keyUri:
            'otpauth://totp/thomas@abc.com?secret=ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ',
        },
      },
    });
    showAppModal({ content: <AddMFAAuthenticatorJourney /> });

    await waitFor(() => {
      expect(
        screen.queryByText(
          'There is a problem with the security code. Try re-entering or resending the code.',
        ),
      ).not.toBeInTheDocument();
    });
    expect(component.baseElement).toMatchSnapshot();
  });
});
