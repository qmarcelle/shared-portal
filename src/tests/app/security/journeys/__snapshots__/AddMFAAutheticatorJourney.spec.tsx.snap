// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticator App Setup should render the screens correctly 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div>
    <div
      class="react-responsive-modal-root"
      data-testid="root"
    >
      <div
        aria-hidden="true"
        class="react-responsive-modal-overlay modal-overlay"
        data-testid="overlay"
        style="animation: react-responsive-modal-overlay-in 300ms;"
      />
      <div
        class="react-responsive-modal-container"
        data-testid="modal-container"
      >
        <div
          aria-modal="true"
          class="react-responsive-modal-modal"
          data-testid="modal"
          role="dialog"
          style="animation: react-responsive-modal-modal-in 300ms;"
          tabindex="-1"
        >
          <div>
            <div
              class="flex flex-col modal-container"
            >
              <div
                class="flex flex-row justify-between modal-header items-center relative"
              >
                <div
                  tabindex="1"
                />
                <img
                  alt="bcbst logo"
                  class="modal-icon modal-header-logo absolute m-auto left-0 right-0"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="size-8 focus-icon"
                  tabindex="0"
                >
                  <img
                    alt="close"
                    class="size-4"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row flex-grow justify-center"
              >
                <div
                  class="flex flex-row justify-center modal-content"
                >
                  <div
                    class="flex flex-col items-center"
                  >
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                    />
                    <h2
                      class="title-2"
                    >
                      Authenticator App Setup
                    </h2>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 body-1 text-center"
                    >
                      Scan the QR Code or manually type the secret key into your authenticator app then enter the security code.
                    </p>
                    <div
                      class="flex flex-col w-[358px]"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-col items-center"
                      >
                        <div
                          style="min-width: 132px; height: 132px;"
                        >
                          <div>
                            <canvas
                              height="128"
                              style="height: 128px; width: 128px;"
                              width="128"
                            />
                          </div>
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <p
                          class="body-1 body-1 font-bold"
                        >
                          Secret Key:
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                        />
                        <p
                          class="body-1"
                        >
                          ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <div
                          class=""
                          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                        />
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <div
                          class="flex flex-col "
                        >
                          <label
                            class="mb-1 "
                            for="input-enter-security-code"
                          >
                            Enter Security Code
                          </label>
                          <input
                            aria-invalid="false"
                            aria-label="Enter Security Code"
                            aria-required="false"
                            class="input  "
                            id="input-enter-security-code"
                            type="text"
                            value=""
                          />
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                      </div>
                      <button
                        aria-label="Confirm"
                        class="button-text primary text-center flex flex-row justify-center items-center min-w-fit inactive font-bold active"
                        type="button"
                      >
                        Confirm
                      </button>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                      />
                    </div>
                    <a
                      aria-label="Cancel"
                      style="display: block;"
                    >
                      <button
                        class="flex flex-row link-container undefined"
                        style="max-width: max-content; height: auto; display: block;"
                        tabindex="-1"
                        type="button"
                      >
                        <p
                          class="link undefined"
                        >
                          Cancel
                        </p>
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row justify-center modal-footer"
              >
                <img
                  alt="bcbst logo"
                  class="modal-icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Authenticator App Setup should render the screens correctly 2`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div>
    <div
      class="react-responsive-modal-root"
      data-testid="root"
    >
      <div
        aria-hidden="true"
        class="react-responsive-modal-overlay modal-overlay"
        data-testid="overlay"
        style="animation: react-responsive-modal-overlay-in 300ms;"
      />
      <div
        class="react-responsive-modal-container"
        data-testid="modal-container"
      >
        <div
          aria-modal="true"
          class="react-responsive-modal-modal"
          data-testid="modal"
          role="dialog"
          style="animation: react-responsive-modal-modal-in 300ms;"
          tabindex="-1"
        >
          <div>
            <div
              class="flex flex-col modal-container"
            >
              <div
                class="flex flex-row justify-between modal-header items-center relative"
              >
                <div
                  tabindex="1"
                />
                <img
                  alt="bcbst logo"
                  class="modal-icon modal-header-logo absolute m-auto left-0 right-0"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="size-8 focus-icon"
                  tabindex="0"
                >
                  <img
                    alt="close"
                    class="size-4"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row flex-grow justify-center"
              >
                <div
                  class="flex flex-row justify-center modal-content"
                >
                  <div
                    class="flex flex-col items-center"
                  >
                    <img
                      alt="success"
                      class="size-[80px]"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                    />
                    <h1
                      class="title-1 title-2"
                    >
                      Authenticator App Setup Complete
                    </h1>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class="flex flex-col w-[358px]"
                    >
                      <div
                        class="flex flex-col items-center"
                      >
                        <p
                          class="body-1 text-center"
                        >
                          The next time you login, you'll generate a code through your authenticator app.
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                        />
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                      />
                      <button
                        aria-label="Done"
                        class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null undefined"
                        type="button"
                      >
                        Done
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row justify-center modal-footer"
              >
                <img
                  alt="bcbst logo"
                  class="modal-icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Authenticator App Setup should render the screens without error when we close the modal previously with error 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div>
    <div
      class="react-responsive-modal-root"
      data-testid="root"
    >
      <div
        aria-hidden="true"
        class="react-responsive-modal-overlay modal-overlay"
        data-testid="overlay"
        style="animation: react-responsive-modal-overlay-in 300ms;"
      />
      <div
        class="react-responsive-modal-container"
        data-testid="modal-container"
      >
        <div
          aria-modal="true"
          class="react-responsive-modal-modal"
          data-testid="modal"
          role="dialog"
          style="animation: react-responsive-modal-modal-in 300ms;"
          tabindex="-1"
        >
          <div>
            <div
              class="flex flex-col modal-container"
            >
              <div
                class="flex flex-row justify-between modal-header items-center relative"
              >
                <div
                  tabindex="1"
                />
                <img
                  alt="bcbst logo"
                  class="modal-icon modal-header-logo absolute m-auto left-0 right-0"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="size-8 focus-icon"
                  tabindex="0"
                >
                  <img
                    alt="close"
                    class="size-4"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row flex-grow justify-center"
              >
                <div
                  class="flex flex-row justify-center modal-content"
                >
                  <div
                    class="flex flex-col items-center"
                  >
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                    />
                    <h2
                      class="title-2"
                    >
                      Authenticator App Setup
                    </h2>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 body-1 text-center"
                    >
                      Scan the QR Code or manually type the secret key into your authenticator app then enter the security code.
                    </p>
                    <div
                      class="flex flex-col w-[358px]"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-col items-center"
                      >
                        <div
                          style="min-width: 132px; height: 132px;"
                        >
                          <div>
                            <canvas
                              height="128"
                              style="height: 128px; width: 128px;"
                              width="128"
                            />
                          </div>
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <p
                          class="body-1 body-1 font-bold"
                        >
                          Secret Key:
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                        />
                        <p
                          class="body-1"
                        >
                          ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <div
                          class=""
                          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                        />
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <div
                          class="flex flex-col "
                        >
                          <label
                            class="mb-1 "
                            for="input-enter-security-code"
                          >
                            Enter Security Code
                          </label>
                          <input
                            aria-invalid="false"
                            aria-label="Enter Security Code"
                            aria-required="false"
                            class="input  "
                            id="input-enter-security-code"
                            type="text"
                            value=""
                          />
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                      </div>
                      <button
                        aria-label="Confirm"
                        class="button-text primary text-center flex flex-row justify-center items-center min-w-fit inactive font-bold active"
                        type="button"
                      >
                        Confirm
                      </button>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                      />
                    </div>
                    <a
                      aria-label="Cancel"
                      style="display: block;"
                    >
                      <button
                        class="flex flex-row link-container undefined"
                        style="max-width: max-content; height: auto; display: block;"
                        tabindex="-1"
                        type="button"
                      >
                        <p
                          class="link undefined"
                        >
                          Cancel
                        </p>
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row justify-center modal-footer"
              >
                <img
                  alt="bcbst logo"
                  class="modal-icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Authenticator App Setup should render the screens without error when we close the modal previously with error 2`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div>
    <div
      class="react-responsive-modal-root"
      data-testid="root"
    >
      <div
        aria-hidden="true"
        class="react-responsive-modal-overlay modal-overlay"
        data-testid="overlay"
        style="animation: react-responsive-modal-overlay-in 300ms;"
      />
      <div
        class="react-responsive-modal-container"
        data-testid="modal-container"
      >
        <div
          aria-modal="true"
          class="react-responsive-modal-modal"
          data-testid="modal"
          role="dialog"
          style="animation: react-responsive-modal-modal-in 300ms;"
          tabindex="-1"
        >
          <div>
            <div
              class="flex flex-col modal-container"
            >
              <div
                class="flex flex-row justify-between modal-header items-center relative"
              >
                <div
                  tabindex="1"
                />
                <img
                  alt="bcbst logo"
                  class="modal-icon modal-header-logo absolute m-auto left-0 right-0"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="size-8 focus-icon"
                  tabindex="0"
                >
                  <img
                    alt="close"
                    class="size-4"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row flex-grow justify-center"
              >
                <div
                  class="flex flex-row justify-center modal-content"
                >
                  <div
                    class="flex flex-col items-center"
                  >
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                    />
                    <h2
                      class="title-2"
                    >
                      Authenticator App Setup
                    </h2>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 body-1 text-center"
                    >
                      Scan the QR Code or manually type the secret key into your authenticator app then enter the security code.
                    </p>
                    <div
                      class="flex flex-col w-[358px]"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-col items-center"
                      >
                        <div
                          style="min-width: 132px; height: 132px;"
                        >
                          <div>
                            <canvas
                              height="128"
                              style="height: 128px; width: 128px;"
                              width="128"
                            />
                          </div>
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <p
                          class="body-1 body-1 font-bold"
                        >
                          Secret Key:
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                        />
                        <p
                          class="body-1"
                        >
                          ZEHLSQVDBQACU44JEF2BGVJ45KHFRDYJ
                        </p>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <div
                          class=""
                          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                        />
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                        <div
                          class="flex flex-col w-full text-field"
                        >
                          <p>
                            Enter Security Code
                          </p>
                          <div
                            class="flex flex-row items-center input  "
                          >
                            <input
                              aria-label="Enter Security Code"
                              type="text"
                              value=""
                            />
                            <div
                              class="cursor-pointer"
                            />
                          </div>
                          <div
                            class="error-container mt-1 p-1"
                          >
                            <div />
                          </div>
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                        />
                      </div>
                      <button
                        aria-label="Confirm"
                        class="button-text primary text-center flex flex-row justify-center items-center min-w-fit inactive font-bold active"
                        type="button"
                      >
                        Confirm
                      </button>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                      />
                    </div>
                    <a
                      aria-label="Cancel"
                      style="display: block;"
                    >
                      <button
                        class="flex flex-row link-container undefined"
                        style="max-width: max-content; height: auto; display: block;"
                        tabindex="-1"
                        type="button"
                      >
                        <p
                          class="link undefined"
                        >
                          Cancel
                        </p>
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row justify-center modal-footer"
              >
                <img
                  alt="bcbst logo"
                  class="modal-icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
